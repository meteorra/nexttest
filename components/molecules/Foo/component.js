import React from 'react';

import atoms from './../../atoms';

const { P } = atoms;

const Foo = () => (
    <div className="Foo">
        <P>Foo component first paragraph</P>
        <P>Foo component second paragraph</P>
    </div>
);

export default Foo;
