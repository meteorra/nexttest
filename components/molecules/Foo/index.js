import Foo from './component';

export default Foo;
