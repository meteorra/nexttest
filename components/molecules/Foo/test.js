import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Foo from './component';

const setUp = {};

describe('Foo', () => {
    beforeAll(() => {
        setUp.render = renderer.create(<Foo />);
        setUp.enzymeWrapper = shallow(<Foo />);
    });
    it('should correspond to the initial template', () => {
        const { render } = setUp;
        expect(render.toJSON()).toMatchSnapshot();
    });
    it('should contain 2 children nodes', () => {
        const { enzymeWrapper } = setUp;
        expect(enzymeWrapper.find('.Foo').children().length).toBe(2);
    });
});
