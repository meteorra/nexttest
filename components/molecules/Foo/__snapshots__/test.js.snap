// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo should correspond to the initial template 1`] = `
<div
  className="Foo"
>
  <p
    className="sc-bdVaJa izxjOo"
  >
    Foo component first paragraph
  </p>
  <p
    className="sc-bdVaJa izxjOo"
  >
    Foo component second paragraph
  </p>
</div>
`;
